test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,196 @@ package e2e
-		packageName := "etcd-test"
-		image := "quay.io/olmtest/catsrc-update-test:related"
-
-		crc := newCRClient(GinkgoT())
-		pmc := newPMClient(GinkgoT())
-
-		source := &v1alpha1.CatalogSource{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       v1alpha1.CatalogSourceKind,
-				APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      sourceName,
-				Namespace: testNamespace,
-				Labels:    map[string]string{"olm.catalogSource": sourceName},
-			},
-			Spec: v1alpha1.CatalogSourceSpec{
-				SourceType: v1alpha1.SourceTypeGrpc,
-				Image:      image,
-			},
-		}
 
-		expectedRelatedImages := map[string]string{
-			"quay.io/coreos/etcd@sha256:3816b6daf9b66d6ced6f0f966314e2d4f894982c6b1493061502f8c2bf86ac84":          "",
-			"quay.io/coreos/etcd@sha256:49d3d4a81e0d030d3f689e7167f23e120abf955f7d08dbedf3ea246485acee9f":          "",
-			"quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2": "",
-		}
+	AfterEach(func() {
+		cleaner.NotifyTestComplete(GinkgoT(), true)
+	})
 
-		source, err := crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
-		require.NoError(GinkgoT(), err)
-		defer func() {
-			require.NoError(GinkgoT(), crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Delete(context.TODO(), source.GetName(), metav1.DeleteOptions{}))
-		}()
-
-		// Wait for package-server to be ready
-		err = wait.Poll(pollInterval, 1*time.Minute, func() (bool, error) {
-			GinkgoT().Logf("Polling package-server...")
-			_, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
-			if err == nil {
-				return true, nil
-			}
-			return false, nil
+	Context("Given a ConfigMap containing package manifest, CSV and CRD", func() {
+		When("A CatalogSource is created using the ConfigMap as catalog source type", func() {
+			It("Package manifest loading can be verified", func() {
+
+				c := newKubeClient()
+
+				// create a simple catalogsource
+				packageName := genName("nginx")
+				stableChannel := "stable"
+				packageStable := packageName + "-stable"
+				manifests := []registry.PackageManifest{
+					{
+						PackageName: packageName,
+						Channels: []registry.PackageChannel{
+							{Name: stableChannel, CurrentCSVName: packageStable},
+						},
+						DefaultChannelName: stableChannel,
+					},
+				}
+
+				crdPlural := genName("ins")
+				crd := newCRD(crdPlural)
+				catalogSourceName := genName("mock-ocs")
+				namedStrategy := newNginxInstallStrategy(genName("dep-"), nil, nil)
+				csv := newCSV(packageStable, testNamespace, "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, namedStrategy)
+				csv.SetLabels(map[string]string{"projected": "label"})
+				csv.Spec.NativeAPIs = []metav1.GroupVersionKind{{Group: "kubenative.io", Version: "v1", Kind: "Native"}}
+				csvJSON, _ := json.Marshal(csv)
```

Comment:
I'm sure this isn't new, but the test shouldn't be ignoring the error value returned from Marshal. A nice thing to add (in a separate PR) would be automation for running static analysis tools like errcheck on every change.